schema/common.json

```diff
@@ -8,6 +8,52 @@
             "type": "object",
             "additionalProperties": true,
             "title": "Headers"
+        },
+        "Otlp": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "protocol": {
+                    "type": "string",
+                    "pattern": "^(http|grpc)\\/(protobuf|json)"
+                },
+                "endpoint": {
+                    "type": "string"
+                },
+                "certificate": {
+                    "type": "string"
+                },
+                "client_key": {
+                    "type": "string"
+                },
+                "client_certificate": {
+                    "type": "string"
+                },
+                "headers": {
+                    "$ref": "#/$defs/Headers"
+                },
+                "compression": {
+                    "type": "string"
+                },
+                "timeout": {
+                    "type": "integer"
+                },
+                "temporality_preference": {
+                    "type": "string"
+                },
+                "default_histogram_aggregation": {
+                    "type": "string"
```

Comment:
Instead of having these metric specific fields in the common definition, we might consider adding a metric specific Otlp exporter type, and use all of syntax to extend it with the metrics specific fields:
```
{
  "allOf": [{ "$ref": "common.json#/#defs/Otlp" }],
  "properties": {
      "temporality_preference": { "type": "string" },
      "default_histogram_aggregation": { "type": "string" },
  }
}
```
I'm not sure of how well supported `allOf` is among the JSON schema validation libraries, but its worth a shot. If its not well supported, we can document that we try to avoid `allOf` in the README.

Reply:
I think I had issues with this when I tried the golang json schema codegen. Worth a try though.

Reply:
`allOf` is not currently supported by the go-jsonschema (library im currently using) https://github.com/omissis/go-jsonschema/issues/6

Doesn't mean we shouldn't use it, but i won't have time to look at it just yet. I could use two different types, or keep the one type w/ all needed fields (since they're optional anyways). Any preference one way or the other?

Reply:
> allOf is not currently supported by the go-jsonschema (library im currently using) https://github.com/omissis/go-jsonschema/issues/6

Dang.. I'm using all of in sampler config in #21 as well. Any ideas how we could structure the schema to constrain the parameters of the various samplers without using allOf?

Reply:
After looking further into allOf, I'm not sure how well it would work for something like the OTLP exporter. `allOf` requires validation against everything defined in the schema, which means that something like `additionalProperties` must be set to `true` in the `Otlp` definition to allow additional fields for `OtlpMetric`.

The following fails validation because OtlpMetric includes additional fields:

```json
      "Otlp": {
            "type": "object",
            "additionalProperties": false,
         ...
         },
        "OtlpMetric": {
            "type": "object",
            "additionalProperties": true,
            "allOf": [{ "$ref": "common.json#/$defs/Otlp" }],
            "properties": {
                "temporality_preference": { "type": "string" },
                "default_histogram_aggregation": { "type": "string" }
            }
        }
```

Making `additionalProperties` true on `Otlp` removes some beneficial validation, for example if someone makes a typo when entering an expected field.

Maybe the best thing is to keep these two definition separate for now?

Reply:
> Maybe the best thing is to keep these two definition separate for now?

That's fair. So that means we need to separately address whether we want to allow usage of allOf in other scenarios, like #21.

Reply:
Managed to get this working through the use of `unevaluatedProperties`, however decided to go back to splitting the otlp metric exporter from the common otlp exporter due to the lack of support for `anyOf` in the tooling. This can always be revisited in the future.